pkg/services/datasources/service/datasource.go

```diff
@@ -231,6 +235,11 @@ func (s *Service) DeleteDataSource(ctx context.Context, cmd *datasources.DeleteD
 
 func (s *Service) UpdateDataSource(ctx context.Context, cmd *datasources.UpdateDataSourceCommand) (*datasources.DataSource, error) {
 	var dataSource *datasources.DataSource
+
+	if err := cmd.Validate(); err != nil {
```

Comment:
it may be better to implement the validation logic here so we don't need to repeat it twice.

Reply:
I agree, a function/method can hold the validation logic here and being shared by both save and update.

Reply:
check it out, could not figure out a better a way of doing it since both func takes different command struct.